openapi.yaml

```diff
@@ -1,279 +1,662 @@
-        example: 10
-      currentPageNumber:
-        type: integer
-        description: Page number of the returned results
-        example: 1
-      currentPageSize:
-        type: integer
-        description: Number of results per page
-        example: 25
-  SelfLink:
-    properties:
-      self:
-        type: string
-        format: uri
-        description: Self-link of current resource
-  PaginationLinks:
-    properties:
-      first:
-        type: string
-        format: uri
-        description: The first page of data
-      last:
-        type: string
-        format: uri
-        description: The last page of data
-      prev:
-        type: string
-        format: uri
-        description: The previous page of data
-      next:
-        type: string
-        format: uri
-        description: The next page of data
-  PetResource:
-    properties:
-      id:
-        type: string
-        description: A unique ID of pet
-      type:
-        $ref: '#/definitions/PetType'
-      links:
-        $ref: '#/definitions/SelfLink'
-      attributes:
-        properties:
-          name:
             type: string
-            description: Name of pet
-            example: Hedwig
-          owner:
+      responses:
+        "200":
+          description: "Successful response"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/ServicesResultObject"
+        "404":
+          description: "Resource not found"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Errors"
+        "500":
+          description: "Internal Server Error"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Errors"
+components:
+  parameters:
+    pretty:
+      name: pretty
+      in: query
+      required: false
+      description: If true, JSON response will be pretty-printed
+      schema:
+        type: boolean
+  securitySchemes:
+    OAuth2:
+      type: oauth2
+      flows: 
+        clientCredentials:
+          tokenUrl: https://api.oregonstate.edu/oauth2/token
+          refreshUrl: https://api.oregonstate.edu/oauth2/token
+          scopes: 
+            full: full access to the API
+  schemas:
+    LocationsResultObject:
+      properties:
+        links:
+          $ref: "#/components/schemas/Links"
+        data:
+          $ref: "#/components/schemas/LocationsResourceObject"
+        included:
+          type: array
+          items:
+            $ref: "#/components/schemas/ServicesResourceObject"
+    LocationsResultObjects:
+      properties:
+        links:
+          $ref: "#/components/schemas/Links"
+        data:
+          type: array
+          items:
+            $ref: "#/components/schemas/LocationsResourceObject"
+        included:
+          type: array
+          items:
+            $ref: "#/components/schemas/ServicesIncluded"
```

Comment:
if the performance of getting a compound document is bad, one option that I think would work is only allowing compound documents with `/locations/{locationId}`, assuming that's something people would find useful

Reply:
By compound documents, are you referring to the plural ResultObjects that references the resource object? I'm not sure if using $ref in the swagger doc will affect the speed of our responses

Reply:
Compound documents meaning a response object from the API that includes all of the resources referenced in the `relationships` keys of the main resource(s) being fetched. [JSON:API docs here](https://jsonapi.org/format/#document-compound-documents). The concern would be that in the API itself we'd need to fetch a bunch of extra resources that we wouldn't have to if we weren't using a compound document.